kart/spatial_filters.py

```diff
@@ -134,36 +155,67 @@ def __init__(self, ref_or_oid):
         super().__init__()
         self.ref_or_oid = ref_or_oid
 
-    def write_config(self, repo):
+    def _resolve_object_contents(self, obj):
+        contents = obj.data.decode("utf-8")
+        parts = self.split_file(contents)
+        return ResolvedSpatialFilterSpec(*parts)
+
+    @functools.lru_cache(maxsize=1)
+    def _resolve_target(self, repo):
+        # Returns a tuple of strings (reference, object_id, ResolvedSpatialFilterSpec).
+        # Returned reference will be None if ref_or_oid is an object-id.
         obj = None
+        oid = self.ref_or_oid
         try:
-            obj = repo[self.ref_or_oid]
+            obj = repo[oid]
         except (KeyError, ValueError):
             pass
 
         if obj is not None:
+            return None, oid, self._resolve_object_contents(obj)
+
+        ref = self.ref_or_oid
```

Comment:
So this function is assuming (AFAICT) that at this point the `self.ref_or_oid` is the name of a ref rather than an OID. Is that true? It seems like it could be an OID of an unfetched object also. Is there any way to tell?

Reply:
It does assume that if it doesn't find the object, it must be a filter - or missing. It could be that it is an unfetched object however. But whether non-existent or unfetched, the end results will be this:
`No spatial filter object was found in the repository at 9d5d8e7 or refs/filters/9d5d8e7` - which is true and describes the problem accurately.

I'll add a TODO to handle unfetched objects here somehow, but it's part of a bigger problem - we can't currently tell if objects are non-existent / corrupted or merely unfetched, and we don't specifically handle unfetched objects anywhere.